### PR TITLE
fix: enabled lint recommended rules

### PR DESCRIPTION
**Pull request by fboucquez**
fix: fixed code related to the new rules.

Fixes https://github.com/symbol/sdk-typescript/issues/36

### CODE REVIEW

File: test/BasicNetworkTest.template.ts [hunk 1]
```diff
@@ -17,12 +17,13 @@
 import { Network } from '@core';
 import { expect } from 'chai';
 
-export const AssertNetwork = (network: Network, expectedName: string, expectedIdentifier: number): void => {
-    it('correct predefined networks are registered', () => {
-        // Assert:
-        expect(network.name).to.be.equal(expectedName);
-        expect(network.identifier).to.be.equal(expectedIdentifier);
-    });
+export const AssertNetwork = (network: Network | undefined, expectedName: string, expectedIdentifier: number): void => {
```
Comment (yilmazbahadir) [inline test/BasicNetworkTest.template.ts:20]: Why do some methods start with a capital letter? Should be having PascalCase for classes only, for this instance, it should be `assertNetwork`?

Comment (fboucquez) [reply test/BasicNetworkTest.template.ts:20]: It should be assertNetwork, I can change it in this pr if no problem.

File: src/core/Address.ts [hunk 2]
```diff
@@ -56,7 +56,11 @@ export abstract class Address {
      * @returns Encoded address tring with separators
```
[the remark this answers is not in the document]
Comment (fboucquez) [reply src/core/Address.ts]: fixed

File: test/BasicNetworkTest.template.ts (hunk 1 above)
Comment (fboucquez) [reply test/BasicNetworkTest.template.ts]: Renamed

**Review by yilmazbahadir — APPROVED** [review]

**Pull request merged**